### ISSUE TITLE
Exception reported: Item is NULL

### ISSUE THREAD

**Issue by some-username-here1** [fakuivan/TF2-Taunts-TF2IDB#40]
I'm not sure what I did wrong when it came to installing the plugin since I did it exactly as the instructions said, but I keep getting this error whenever I select any taunt:

```
[TF2ITEMS] ---------------------------------------
[TF2ITEMS] >>> szClassname = tf_wearable_vm
[TF2ITEMS] >>> iItemDefinitionIndex = 167
[TF2ITEMS] >>> iEntityQuality = 6
[TF2ITEMS] >>> iEntityLevel = 1
[TF2ITEMS] ---------------------------------------
[SM] Exception reported: Item is NULL. File a bug report if you are sure you set all the data correctly. (Try the FORCE_GENERATION flag.)
[SM] Blaming: tf2_taunts_tf2idb.smx
[SM] Call stack trace:
[SM]   [0] TF2Items_GiveNamedItem
[SM]   [1] Line 58, /home/travis/build/fakuivan/TF2-Taunts-TF2IDB/addons/sourcemod/scripting/include/tf2_taunts_tf2idb/taunt_enforcer.inc::CTauntEnforcer.ForceTaunt
[SM]   [2] Line 222, /home/travis/build/fakuivan/TF2-Taunts-TF2IDB/addons/sourcemod/scripting/include/tf2_taunts_tf2idb/tf2_taunts_tf2idb.inc::CheckAndTaunt
[SM]   [3] Line 282, /home/travis/build/fakuivan/TF2-Taunts-TF2IDB/addons/sourcemod/scripting/tf2_taunts_tf2idb.sp::MenuHandler_TauntsSelfMenu
```

**Comment by fakuivan** [comment]
This might be a tf2items issue. Have you tried updating gamedata files for tf2items?

**Comment by some-username-here1** [comment]
Well yeah, my tf2items is up-to-date, given that I got the latest from [here](https://builds.limetech.io/?project=tf2items)

But if there's any other place I can get a more recent version of tf2items, since I don't want to entirely build tf2items from the github page itself, please let me know

**Comment by fakuivan** [comment]
iirc that is indeed where asher pushes the new versions, or used to be anyways. What item schema api are you using, and how old is the database if you're using tf2idb? Maybe we're giving erroneous data to tf2items.

**Comment by Dragonissa** [comment]
Having the same issue. Everything up to date tho. 

Might be related to how they get the item in the latest versions. Im gonna try going back some versions and see if it changes anything.

Edit:


Went back all the way to TF2Items 1.5.3-240 and even the first versions of this plugin. No change.

**Comment by fakuivan** [comment]
Going back to older versions might not be what you want, since usually updates incorporate changes in how these extensions interact with new game code, with no stable interface, valve might break them intentionally or unintentionally with updates. This reactionary nature is what breaks servers from time to time, with needed gamedata or extension updates. 
Do you recall this having stopped working at some specific time or update to the game, sourcemod or any other component? I would find it strange for tf2ii to introduce any breaking changes to the API at this point.
<sub><sup>or maybe valve finally took the time to break the hacks</sup></sub>

**Comment by Dragonissa** [comment]
I don't use the tf2ii version and i haven't used this plugin before either. 
Literally installed it yesterday since TauntEm is even more broken.

**Comment by fakuivan** [comment]
I see. Are you using an updated tf2idb database?

**Comment by Dragonissa** [comment]
Yes, updated yesterday when i installed it. I downright think there was a change in TF2Items that broke the function.

**Comment by fakuivan** [comment]
If that were the case then using an older version would fix the issue. I'm guessing there's something more here. Right now I don't have a testing server or scripting environment in hand to test this. I'll probably have time to test this midweek. In the meantime I'll ask around in the sm discord.

**Comment by Dragonissa** [comment]
Ill take a look at it myself, eventually.

**Comment by Dragonissa** [comment]
Just went through all popular taunt plugins and it seems to be a common problem. There must be a way since skial is still using it just fine.

**Comment by LemonatedCat** [comment]
I can confirm this issue aswell on my end

**Comment by LemonatedCat** [comment]
```L 03/31/2021 - 16:21:53: [SM]   [3] Line 282, /home/travis/build/fakuivan/TF2-Taunts-TF2IDB/addons/sourcemod/scripting/tf2_taunts_tf2idb.sp::MenuHandler_TauntsSelfMenu
L 03/31/2021 - 16:21:57: [TF2ITEMS] ---------------------------------------
L 03/31/2021 - 16:21:57: [TF2ITEMS] >>> szClassname = tf_wearable_vm
L 03/31/2021 - 16:21:57: [TF2ITEMS] >>> iItemDefinitionIndex = 1117
L 03/31/2021 - 16:21:57: [TF2ITEMS] >>> iEntityQuality = 6
L 03/31/2021 - 16:21:57: [TF2ITEMS] >>> iEntityLevel = 1
L 03/31/2021 - 16:21:57: [TF2ITEMS] ---------------------------------------
L 03/31/2021 - 16:21:57: [SM] Exception reported: Item is NULL. File a bug report if you are sure you set all the data correctly. (Try the FORCE_GENERATION flag.)
L 03/31/2021 - 16:21:57: [SM] Blaming: tf2_taunts_tf2idb.smx
L 03/31/2021 - 16:21:57: [SM] Call stack trace:
L 03/31/2021 - 16:21:57: [SM]   [0] TF2Items_GiveNamedItem
L 03/31/2021 - 16:21:57: [SM]   [1] Line 58, /home/travis/build/fakuivan/TF2-Taunts-TF2IDB/addons/sourcemod/scripting/include/tf2_taunts_tf2idb/taunt_enforcer.inc::CTauntEnforcer.ForceTaunt
L 03/31/2021 - 16:21:57: [SM]   [2] Line 222, /home/travis/build/fakuivan/TF2-Taunts-TF2IDB/addons/sourcemod/scripting/include/tf2_taunts_tf2idb/tf2_taunts_tf2idb.inc::CheckAndTaunt
L 03/31/2021 - 16:21:57: [SM]   [3] Line 282, /home/travis/build/fakuivan/TF2-Taunts-TF2IDB/addons/sourcemod/scripting/tf2_taunts_tf2idb.sp::MenuHandler_TauntsSelfMenu

```

**Comment by LemonatedCat** [comment]
I've found the fix, a previous issue (or maybe on tf2idb) suggested to remove "NOT NULL" from the db generation script on line 78, undoing that change makes it work correctly again.

**Comment by Dragonissa** [comment]
> I've found the fix, a previous issue (or maybe on tf2idb) suggested to remove "NOT NULL" from the db generation script on line 78, undoing that change makes it work correctly again.

Does it work then? That sounds more like a way to fix the issue that an item in the db cant be null, but not the fact that its even there.

Line 78 also doesnt look like the correct line. Can you link it on github?

https://github.com/FlaminSarge/tf2idb/issues/19

Delete the "NOT NULL" in line 44.

![image](https://user-images.githubusercontent.com/4233773/113194040-f7fcfd00-9260-11eb-93d6-87717fd38d4c.png)

![image](https://user-images.githubusercontent.com/4233773/113194071-00edce80-9261-11eb-87fe-e26604ac3828.png)



Edit:

~~Tried it and nothing changed.~~
Server restart it required. Can be marked as solved-

@some-username-here1

**Comment by fakuivan** [comment]
Lmao thanks @fakuivan from the past. I'll mark this as closed then.

**Comment by Dragonissa** [comment]
you might wanna update the python script.

**Comment by fakuivan** [comment]
@Dragonisser the python script has already been fixed a long time ago https://github.com/FlaminSarge/tf2idb/commit/3ae9e20300b9ca8982d0f40cb6a56dcfa1c33ce6

**Comment by Dragonissa** [comment]
> @Dragonisser the python script has already been fixed a long time ago [FlaminSarge/tf2idb@3ae9e20](https://github.com/FlaminSarge/tf2idb/commit/3ae9e20300b9ca8982d0f40cb6a56dcfa1c33ce6)

Which was never updated on AlliedModders, for whatever reason. Last updated was 2 days before that fix.
